gtest_parallel_tests.py

```diff
@@ -159,35 +148,106 @@ def test_run_task_basic(self):
 
     self.assertEqual(task_manager.global_exit_code, 1)
 
-  @unittest.skipIf(sys.platform == 'win32', 'non-Windows specific')
-  def test_get_save_file_path_unix(self):
-    self.assertEqual(os.path.join(os.path.expanduser('~'),
-                                  '.cache', 'gtest-parallel'),
-                     gtest_parallel.get_save_file_path())
 
-    with guard_environ('XDG_CACHE_HOME', os.getcwd()):
-      self.assertEqual(os.path.join(os.getcwd(), 'gtest-parallel'),
-                       gtest_parallel.get_save_file_path())
+@contextlib.contextmanager
+def guard_environ(var, val):
+  try:
+    old_val = os.environ.get(var)
+    if val is None:
+      if old_val is not None:
+        del os.environ[var]
+    else:
+      os.environ[var] = val
+    yield old_val
+  finally:
+    if old_val is None:
+      if val is not None:
+        del os.environ[var]
+    else:
+      os.environ[var] = old_val
+
+
+@contextlib.contextmanager
+def guard_temp_dir():
+  try:
+    temp_dir = tempfile.mkdtemp()
+    yield temp_dir
+  finally:
+    shutil.rmtree(temp_dir)
 
-    with guard_environ('XDG_CACHE_HOME', os.path.realpath(__file__)):
-      self.assertEqual(os.path.join(os.path.expanduser('~'),
-                                    '.gtest-parallel-times'),
-                       gtest_parallel.get_save_file_path())
 
-  @unittest.skipUnless(sys.platform == 'win32', 'Windows specific')
+@contextlib.contextmanager
+def guard_temp_subdir(temp_dir, *path):
+  assert path, 'Path should not be empty'
+
+  try:
+    temp_subdir = os.path.join(temp_dir, *path)
+    os.makedirs(temp_subdir)
+    yield temp_subdir
+  finally:
+    shutil.rmtree(os.path.join(temp_dir, path[0]))
+
+
+@contextlib.contextmanager
+def guard_patch_module(import_name, new_val):
```

Comment:
All the guards implement contextmanager protocol. They don't test anything, they only guard that global changes (which can influence other tests) are correctly reverted. I can rename them but test_ is not a good candidate for this name either.